common/src/main/java/com/microsoft/identity/common/nativeauth/internal/controllers/NativeAuthMsalController.kt

```diff
@@ -319,6 +328,84 @@ class NativeAuthMsalController : BaseNativeAuthController() {
         }
     }
 
+    fun signInSubmitChallenge(parameters: MFASubmitChallengeCommandParameters): MFASubmitChallengeCommandResult {
+        LogSession.logMethodCall(
+            tag = TAG,
+            correlationId = parameters.getCorrelationId(),
+            methodName = "${TAG}.signInSubmitCode"
+        )
+
+        try {
+            // Add default scopes
+            val mergedScopes: List<String> = addDefaultScopes(parameters.scopes)
+
+            val parametersWithScopes =
+                CommandUtil.createMFASubmitChallengeCommandParametersWithScopes(
+                    parameters,
+                    mergedScopes
+                )
+
+            val oAuth2Strategy = createOAuth2Strategy(parameters)
+
+            val tokenApiResult = performSubmitChallengeTokenRequest(
+                oAuth2Strategy = oAuth2Strategy,
+                parameters = parametersWithScopes
+            )
+            return when (tokenApiResult) {
+                is SignInTokenApiResult.Success -> {
+                    saveAndReturnTokens(
+                        oAuth2Strategy = oAuth2Strategy,
+                        parametersWithScopes = parametersWithScopes,
+                        tokenApiResult = tokenApiResult
+                    )
+                }
+                is SignInTokenApiResult.MFARequired -> {
+                    SignInCommandResult.MFARequired(
```

Comment:
I'm wondering if the "MFARequired" status is implemented as intended. I thought we would expect "MFA required" result only after entering a password, not a code. In this case the "SignInCodeRequiredState" needs to be updated too, and if I'm not wrong this wasn't part of the technical design.  Please, correct me if I'm wrong here

Reply:
That limitation is currently handled in the interface layer, but indeed moving it here makes it cleaner.